### ISSUE TITLE
Add branch compatible to Yearn.Finance v2 DAI Vault 

### ISSUE THREAD

**Issue by Deepcryptodive** [Good-Ghosting/goodghosting-smart-contracts#64]
Soon, Yearn.Finance will launch their v2 Vaults ([Github](https://github.com/iearn-finance/yearn-vaults) & [spec](https://github.com/iearn-finance/yearn-vaults/blob/master/SPECIFICATION.md)), including one accepting DAI.

Since users have expressed interest in having "higher risk, higher reward" saving pools, it makes sense to create a branch that allows deploying GoodGhosting games that use the new DAI vault (of Yearn Finance) to generate yield.

The Yearn DAI vault works like this:
- if you `deposit()` DAI into the Yearn DAI Vault, you get an ERC-20 token representing your shares in the vault  (e.g. [yvDAI](https://etherscan.io/token/0x1b048ba60b02f36a7b48754f4edf7e1d9729ebc9))
- the $ value of these vault share tokens goes up over time, which can be tracked via `pricePerShares()` 
- you can exit the vault any time: by redeeming your vault shares for DAI by calling  `withdraw()`

_This is somewhat similar to Aave (mint() and redeem() functions), with the excepting that the token balance of the vault shares does not increase in number like aDAI does! Only the number of DAI that a vault share can be redeemed for, increases._ 

**Pros**
- **Gas savings**
Deposits of DAI will only [cost](https://etherscan.io/tx/0xaec13a3b3404a744c18707dfd1bfa462dfb057ca2bb99fb251cbe5659d332466) ~100k of gas. 
Compare this to ~240k in gas for depositing into Aave v1 (which we currently use), or ~150k of gas for depositing into Aave2. 
- **(Potentially) higher interest rates than on Aave**
Historically, the interest rate on Yearn Vaults [has been](https://yearn.finance/stats) over double that of Aave. This would make the saving pool allot more attractive. _No guarantees though._

**Cons**
- **Slightly increased smart contract risk** of the funds, compared to Aave which has been heavily battle-tested.
- No ability to test on Kovan?  (_as far as I know, Yearn developers use a fork of mainnet)_
- If an 'emergency shutdown' is triggered, no new deposits in the Vault are possible (even though withdrawals are no problem). This may stall the game.
- They have not rolled out the Vault to the public - still in closed testing phase (on mainnet).
The latest contract can be found here: https://etherscan.io/address/0x1b048ba60b02f36a7b48754f4edf7e1d9729ebc9#code
_Note: for testing purposes, you need to hold YFI to be able to interact with it._

**Timing**
Probably best to implement this only after completing the batch deposit functionality (cf #54 ) and refactoring of segments. 

**tl;dr This will enable the GoodGhosting savings pool to get an optimal risk-adjusted yield.**

_You may wonder: why not use the current DAI Vault of Yearn Finance (v1)? Well, the current vault has a 0.5% withdrawal fee. This fee has been removed in the v2 vault, and replaced with a performance and management fee. Also, the v2 Vault is expected to have higher returns, due to being able to run multiple strategies at the same time._
